mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -451,6 +450,38 @@ pub async fn lp_init_continue(ctx: MmArc) -> MmInitResult<()> {
             .map_to_mm(MmInitError::ErrorSqliteInitializing)?;
         init_and_migrate_sql_db(&ctx).await?;
         migrate_db(&ctx)?;
+        // Initialize the global and wallet directories and databases since these are constants over the lifetime of KDF.
+        #[cfg(all(feature = "new-db-arch", not(target_arch = "wasm32")))]
```

Comment:
Why do you need `not(target_arch = "wasm32")` here when it's already there https://github.com/KomodoPlatform/komodo-defi-framework/blob/b1027d2b2f25531761efed764c0eb002c6769487/mm2src/mm2_main/src/lp_native_dex.rs#L441-L443
Also, why didn't you make this block a function?

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2378/commits/75d9cf0715797d56d86573f4b51f44786401c085

kept the ensure-writable funcs and so since these use methods from lp_native_dex.